### PR TITLE
CRM-18019: Fix regression with views display of custom fields

### PR DESCRIPTION
**Pull request by JKingsnorth**
- [CRM-18019: Display of custom checkbox field values is broken on event confirmation page](https://issues.civicrm.org/jira/browse/CRM-18019)

### CODE REVIEW

**Comment by JKingsnorth** [comment]
Jenkins, re-test this please

File: CRM/Core/BAO/CustomField.php [hunk 1]
```diff
@@ -1223,7 +1223,18 @@ public static function getDisplayValueCommon(
       case 'AdvMulti-Select':
       case 'Multi-Select':
         if (is_array($value)) {
-          $checkedData = $value;
+          if ($html_type == 'CheckBox') {
+            $newData = array();
+            foreach ($checkedData as $k => $v) {
```
Comment (jitendrapurohit) [inline CRM/Core/BAO/CustomField.php]: @JKingsnorth The build fails due to the undefined variable `$checkedData` used here as the initialisation is removed before the if stmt.

**Comment by jitendrapurohit** [comment]
Actually, this issue should be fixed by simply calling `CRM_Utils_Array::formatArrayKeys()` just after checking `$value` is an array.

Eg:

    if (is_array($value)) {
      if ($html_type == 'CheckBox') {
        CRM_Utils_Array::formatArrayKeys($value);
      }
    }

File: CRM/Core/BAO/CustomField.php (hunk 1 above)
Comment (JKingsnorth) [inline CRM/Core/BAO/CustomField.php]: Doh, obviously. Thanks.

**Comment by JKingsnorth** [comment]
@jitendrapurohit Great, thanks for suggesting that. I hadn't seen that function before (to be honest I wouldn't have guessed that's what it does anyway ;])

Your proposed solution does seem to fix the bug. Much cleaner. @rocxa can you also confirm this new patch works for you?

**Pull request merged**